File: upwork-intelligence/merge-batch.mjs
#!/usr/bin/env node
import fs from 'fs';
import path from 'path';
const rawPath = path.join(path.dirname(new URL(import.meta.url).pathname), '.run-raw.json');
const batch = JSON.parse(fs.readFileSync(0, 'utf8'));
const data = JSON.parse(fs.readFileSync(rawPath, 'utf8'));
for (const { keyword, jobs, error } of batch) {
  const entry = { keyword, jobs: jobs || [], error: error || null };
  const i = data.searches.findIndex((s) => s.keyword === keyword);
  if (i >= 0) data.searches[i] = entry;
  else data.searches.push(entry);
  if (error) data.errors = [...new Set([...(data.errors || []), keyword])];
}
fs.writeFileSync(rawPath, JSON.stringify(data));

File: upwork-intelligence/_merge_batch.py
#!/usr/bin/env python3
import json
import sys
from pathlib import Path

BASE = Path(__file__).resolve().parent
raw_path = BASE / "_raw_run.json"
raw = json.loads(raw_path.read_text())
batch = json.loads(sys.stdin.read())
if isinstance(batch, dict) and "keyword" in batch:
    batch = [batch]
for item in batch:
    raw["searches"].append(item)
raw["meta"]["keywordsCompleted"] = len(raw["searches"])
raw_path.write_text(json.dumps(raw))
print(len(raw["searches"]))
